include/pybind11/pytypes.h

```diff
@@ -478,6 +478,42 @@ inline handle get_function(handle value) {
     return value;
 }
 
+// Reimplementation of python's dict helper functions
+
+// copied from cpython _PyDict_GetItemStringWithError
+inline PyObject * dict_getitemstring(PyObject *v, const char *key)
```

Comment:
nit Can you summarize rationale for this? (and also for traceability?)
e.g. "We should use `*WithError` flavors to ensure we don't accidentally shadow other signals like KeyboardInterrupt (#2862)" or sthg like that? (either ref'ing your issue, or https://bugs.python.org/issue35459)?

Reply:
Done.